src/backend/src/router/triage_agent_router.py

```diff
@@ -51,46 +46,82 @@ def triage_agent_router(
-
-                        # Load the agent response into a JSON
-                        if msg.role == "assistant" :
-                            try:
-                                # Attempt to parse the agent response as JSON
-                                data = json.loads(last_text.text.value)
-                                parsed_result = parse_response(data)
-                                return parsed_result
-                            except json.JSONDecodeError as e:
-                                _logger.error(f"Error decoding JSON on attempt {attempt}: {e}")
-                                _logger.error(f"Raw JSON string: {last_text.text.value}")
-
-                                # If JSON parsing fails, handle retries or raise an error if max retries reached
-                                if attempt == max_retries:
-                                    raise RuntimeError(f"JSON parsing failed after {max_retries} attempts.")
-                                else:
-                                    # Exit the inner loop to retry agent run
-                                    _logger.info(f"Retrying agent run due to JSON parsing error... Attempt {attempt + 1}/{max_retries}")
-                                    break
-        
-            # If run fails, handle retries or raise an error if max retries reached
-            elif attempt == max_retries:
-                _logger.error(f"Run failed after {max_retries} attempts: {run.last_error}")
-                raise RuntimeError()
-            else:
-                _logger.warning(f"Run failed on attempt {attempt}: {run.last_error}. Retrying...")
+        return thread
 
+    def handle_successful_run(
+        thread: AgentThread, 
+        attempt: int
+    ) -> dict:
+        """
+        Helper function to handle a successful agent run
+        """
+        # Parse the agent response from the successful run
+        _logger.info(f"Agent run succeeded on attempt {attempt}.")
+        messages = agents_client.messages.list(thread_id=thread.id, order=ListSortOrder.ASCENDING)
+        for msg in messages:
+            # Grab the last text message from the assistant
+            if msg.text_messages and msg.role == "assistant":
+                last_text = msg.text_messages[-1]
+                _logger.info(f"{msg.role}: {last_text.text.value}")
+
+                # Load the agent response into a JSON
+                try:
+                    data = json.loads(last_text.text.value)
+                    _logger.info(f"Agent response parsed successfully: {data}")
+                    parsed_result = parse_response(data)
+                    return parsed_result
+                
+                # Raise error if agent response cannot be parsed
+                except Exception as e:
+                    _logger.error(f"Agent response failed with error: {e}")
+                    raise ValueError(f"Failed to parse agent response: {e}")
```

Comment:
There's a chance here that we return None,

can you put a guard at the bottom of handle_successful_run that raises an exception so the caller can handle that?

Reply:
e.g. what if there's nothing in messages? what if there's no messages from the assistant?